src/IO/WriteBufferFromS3.cpp

```diff
@@ -101,13 +94,18 @@ void WriteBufferFromS3::finalizeImpl()
         writePart();
         completeMultipartUpload();
     }
-
-    finalized = true;
 }
 
 WriteBufferFromS3::~WriteBufferFromS3()
 {
-    finalizeImpl();
+    try
+    {
+        next();
+    }
+    catch (...)
+    {
+        tryLogCurrentException(__PRETTY_FUNCTION__);
+    }
```

Comment:
But this will hide write error in case of exception, the idea of #19451 is to move such code into the callers (although it is quite a challenge)

Reply:
Unfortunately, I don't get this change, did you find a way to process two exceptions simultaneously? Is `std::terminate` a better thing than logging here?

Reply:
>Is std::terminate a better thing than logging here?

Yes, since ignoring write error is unacceptable, it is better to terminate.

Basically it should not be possible if all callers calls finalize(), but if some code path will miss it, then the error may be ignored and the user will not get any error (yes there will be an error in error log, but this is not enough)